### PR TITLE
wfprev-143 Edit project: UI

### PR DESCRIPTION
**Pull request by yzlucas**
image
image

### CODE REVIEW

File: client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    // Add a marker at the project's coordinates
+    L.marker([this.sampleData.coordinats[0], this.sampleData.coordinats[1]]).addTo(this.map);
+
+    // Bind a popup to the marker
+    // marker.bindPopup('Project Location: ' + this.sampleData.projectName).openPopup();
+  }
+
+  onSave(): void {
+    if (this.detailsForm.valid) {
+      console.log('Form Data:', this.detailsForm.value);
```
Comment (agordon-vivid) [inline client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts:125]: plz remove

**Review by agordon-vivid — CHANGES_REQUESTED** [review]
Nice work.  A couple of small items that need to be addressed.

File: client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts (hunk 1 above)
Comment (yzlucas) [reply client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts]: removed

File: client/wfprev-war/src/main/angular/src/app/app.routing.ts [hunk 2]
```diff
@@ -26,6 +26,12 @@ const PANEL_ROUTES: Routes = [
     component: ErrorPageComponent,
     pathMatch: 'full',
   },
+  {
+    path: ResourcesRoutes.EDIT_PROJECT,
+    loadChildren: () =>
+      import('src/app/components/edit-project.module').then(m => m.EditProjectModule),
+  },
+  { path: '', redirectTo: ResourcesRoutes.MAP, pathMatch: 'full' }, // Default route to map
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/src/app/app.routing.ts:34]: Please add the authentication check **canActivate: [PrevAuthGuard]** like the list and map pages above

File: client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.html [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      <div class="form-item">
+        <label>BC Parks District</label>
+        <select id="bcParksDistrict" formControlName="bcParksDistrict">
+          <option value="bcParksDistrict1">bcParksDistrict1</option>
+          <option value="bcParksDistrict2">bcParksDistrict2</option>
+          <option value="bcParksDistrict3">bcParksDistrict3</option>
+        </select>
+      </div>
+  
+      <!-- <div class="form-item full-width">
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.html:137]: Remove this commented code?

File: client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    totalPlannedProjectSizeHa: 500.0,
+    totalPlannedCostPerHectare: 200.0,
+    totalActualAmount: 0.0,
+    isMultiFiscalYearProj: false,
+    forestRegionOrgUnitId: 1001,
+    forestDistrictOrgUnitId: 2001,
+    fireCentreOrgUnitId: 3001,
+    bcParksRegionOrgUnitId: 4001,
+    bcParksSectionOrgUnitId: 5001,
+    coordinats:[48.407326,-123.329773],
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-details.component.ts:50]: Typo here, should be **coordinates**

File: client/wfprev-war/src/main/angular/src/app/components/list-panel/projects-list/projects-list.component.html [hunk 5]
```diff
@@ -53,7 +53,7 @@
                 <img alt="detail-icon" class="icon" src="/assets/detail-icon.svg" />
                 <span>Details</span>
               </div>
-              <div class="action-item">
+              <div class="action-item" (click)="editProject(project)" #stopPropagation>
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/src/app/components/list-panel/projects-list/projects-list.component.html:56]: What is #stopPropagation? I don't see it referenced elsewhere

**Review by ssylver93 — CHANGES_REQUESTED** [review]
Looking good, I left a couple of suggestions

**Review by agordon-vivid — DISMISSED** [review]

File: client/wfprev-war/src/main/angular/src/app/components/list-panel/projects-list/projects-list.component.html (hunk 5 above)
Comment (yzlucas) [reply client/wfprev-war/src/main/angular/src/app/components/list-panel/projects-list/projects-list.component.html:56]: I use it to stop the click effect on expanding the panel. But I just moved this to typescript function "event.stopPropagation();" to handle it

**Comment by yzlucas** [comment]
> Looking good, I left a couple of suggestions

updated. 
Fixing some of the failing unit tests now

File: client/wfprev-war/src/main/angular/src/app/app.routing.ts [hunk 6]
```diff
@@ -26,10 +26,23 @@ const PANEL_ROUTES: Routes = [
     component: ErrorPageComponent,
     pathMatch: 'full',
   },
-
-  { path: '', 
-    redirectTo: ResourcesRoutes.MAP, 
-    pathMatch: 'full' } // Default route to map
+  {
+    path: ResourcesRoutes.EDIT_PROJECT,
+    loadChildren: () =>
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/src/app/app.routing.ts:31]: Shouldn't the authentication be getting checked on the edit project page as well?

Comment (yzlucas) [reply client/wfprev-war/src/main/angular/src/app/app.routing.ts:31]: added

**Pull request merged**